retromock/src/main/java/co/infinum/retromock/Retromock.java

```diff
@@ -223,8 +215,59 @@ private Retromock(final Retrofit retrofit,
     }
     final T delegate = factory.create();
 
-    return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class[] {service},
+    return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class[]{service},
       new InvocationHandler() {
+
+        private boolean isKotlinSuspendFunction;
+        private boolean continuationWantsResponse;
```

Comment:
This isn't okay. InvocationHandler shouldn't have any state variables.
Imagine there are two calls to this in parallel.
Call1 in Thread1 sets isKotlinSuspendFunction = false. Right after it, Call2 from Thread2 sets isKotlinSuspendFunction = true. After few instructions more Call1 is at the end of the invoke functions and it checks should it suspend the function. As isKotlinSuspendFunction is true it would try to suspend unsuspendable function.